tests/profiles/universal_test.py

```diff
@@ -486,19 +486,59 @@ def test_check_required_tables():
     check = CheckTester(universal_profile,
                         "com.google.fonts/check/required_tables")
 
-    required_tables = ["cmap", "head", "hhea", "hmtx",
+    REQUIRED_TABLES = ["cmap", "head", "hhea", "hmtx",
                        "maxp", "name", "OS/2", "post"]
-    optional_tables = ["cvt ", "fpgm", "loca", "prep",
+
+    OPTIONAL_TABLES = ["cvt ", "fpgm", "loca", "prep",
                        "VORG", "EBDT", "EBLC", "EBSC",
                        "BASE", "GPOS", "GSUB", "JSTF",
-                       "gasp", "hdmx", "kern", "LTSH",
-                       "PCLT", "VDMX", "vhea", "vmtx"]
+                       "gasp", "hdmx", "LTSH", "PCLT",
+                       "VDMX", "vhea", "vmtx", "kern"]
+
     # Our reference Mada Regular font is good here
     ttFont = TTFont(TEST_FILE("mada/Mada-Regular.ttf"))
+    cff_ft = TTFont(TEST_FILE("source-sans-pro/OTF/SourceSansPro-Black.otf"))
+    cff2_ft = TTFont(TEST_FILE("source-sans-pro/VAR/SourceSansVariable-Italic.otf"))
 
     # So it must PASS the check:
-    assert_PASS(check(ttFont),
-                'with a good font...')
+    msg = assert_PASS(check(ttFont), 'with a good font...')
+    assert msg == "Font contains all required tables."
+
+    msg = assert_results_contain(check(ttFont), INFO, "optional-tables")
+    for tag in {"loca", "GPOS", "GSUB"}:
+        assert tag in msg
+
+    # CFF ------
+    msg = assert_PASS(check(cff_ft), 'with a good font...')
+    assert msg == "Font contains all required tables."
+
+    msg = assert_results_contain(check(cff_ft), INFO, "optional-tables")
+    for tag in {"BASE", "GPOS", "GSUB"}:
+        assert tag in msg
+
+    del cff_ft.reader.tables["CFF "]
+    msg = assert_results_contain(check(cff_ft), FAIL, "required-tables")
+    assert "CFF " in msg
+
+    # CFF2 ------
+    msg = assert_PASS(check(cff2_ft), 'with a good font...')
+    assert msg == "Font contains all required tables."
+
+    msg = assert_results_contain(check(cff2_ft), INFO, "optional-tables")
+    for tag in {"BASE", "GPOS", "GSUB"}:
+        assert tag in msg
+
+    del cff2_ft.reader.tables["CFF2"]
+    msg = assert_results_contain(check(cff2_ft), FAIL, "required-tables")
+    assert "CFF2" in msg
+
+    del cff2_ft.reader.tables["STAT"]
+    msg = assert_results_contain(check(cff2_ft), FAIL, "required-tables")
+    assert "STAT" in msg
+
+    del cff2_ft.reader.tables["fvar"]
+    msg = assert_results_contain(check(cff2_ft), FAIL, "required-tables")
+    assert "CFF " in msg
```

Comment:
This one is not immediately clear to the reader, so please add a comment here explaining why this must be the case (and also a similar comment above, for the `assert "CFF2" in msg` code-test).

Reply:
👍

Reply:
Comments added in 3e08b6911ab98d158b2dca3a09e2b5eb5a62b106